File: chat-service/src/main/java/com/wanderbee/chatservice/model/ChatRoom.java
package com.wanderbee.chatservice.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.index.Indexed;

import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "chat_rooms")
public class ChatRoom {

    @Id
    private String id;

    private String name;

    private boolean isGroup;

    @Builder.Default
    @Indexed
    private List<String> participantIds = new ArrayList<>();

    /** Snapshot of the most recent message – populated after each send */
    private ChatMessage lastMessage;
}

File: app/src/main/java/com/example/wanderbee/data/remote/models/chat/ChatRoom.kt
package com.example.wanderbee.data.remote.models.chat

/**
 * Matches the backend ChatRoom MongoDB document.
 */
data class ChatRoom(
    val id: String = "",
    val name: String = "",
    val isGroup: Boolean = false,
    val participantIds: List<String> = emptyList(),
    val lastMessage: ChatMessage? = null
)